core/darwin/src/internal/TimeZoneNative.kt

```diff
@@ -12,10 +12,18 @@ import kotlinx.datetime.internal.*
 import platform.Foundation.*
 
 internal actual fun timeZoneById(zoneId: String): TimeZone =
-    RegionTimeZone(tzdb.getOrThrow().rulesForId(zoneId), zoneId)
+    runCatching { RegionTimeZone(tzdb.getOrThrow().rulesForId(zoneId), zoneId) }
+        .getOrElse { timeZoneByIdFoundation(zoneId) }
+
+internal fun timeZoneByIdFoundation(zoneId: String): TimeZone =
+    RegionTimeZone(TimeZoneRulesFoundation(zoneId), zoneId)
 
 internal actual fun getAvailableZoneIds(): Set<String> =
-    tzdb.getOrThrow().availableTimeZoneIds()
+    runCatching { tzdb.getOrThrow().availableTimeZoneIds() }
+        .getOrElse { getAvailableZoneIdsFoundation() }
```

Comment:
Can `availableTimeZoneIds` even reasonably fail? I think it will just return broken/incomplete data, even if the fallback contains some timezone identifiers. One more reason to consistently stick to either the filesystem or the fallback.